astropy/table/table.py

```diff
@@ -3644,12 +3649,33 @@ def _convert_col_for_table(self, col):
             qcol = q_cls(col.data, col.unit, copy=False)
             qcol.info = col.info
             qcol.info.indices = col.info.indices
+            if np.issubdtype(getattr(col, 'dtype', None), np.int):
+                self._dtype_before_conversion = col.dtype
             col = qcol
         else:
             col = super()._convert_col_for_table(col)
 
         return col
 
+    def replace_column(self, name, col, copy=True):
+        super().replace_column(name, col, copy)
+        if self._dtype_before_conversion is not None:
+            warnings.warn("dtype is converted to float for Quantity column '{}'. "
+                          "See https://docs.astropy.org/en/stable/api/astropy.table.QTable.html"
+                          "".format(col.info.name))
```

Comment:
```suggestion
            warnings.warn(f"dtype is converted to float for Quantity column '{col.info.name}'. "
                          "See https://docs.astropy.org/en/stable/api/astropy.table.QTable.html",
                          AstropyUserWarning)
```

Reply:
Remind me: is `AstropyUserWarning` just shown once by default, or every time? I'm not sure we want the every time behaviour here - people will just stop looking at warnings if they happen too frequently...